### PR TITLE
Add puppet manifest syntax support - use .pp extension

### PR DESCRIPTION
**Pull request by ghost**
(.pp is currently used for Pascal highlighting)

Support for puppet manifest highlighting was kindly
provided by Sergey Zhuga, original is here:
https://github.com/scrill/puppet-syntax-mcedit

Mr. Zhuga agreed with inclusion in mainstream MC:
https://github.com/scrill/puppet-syntax-mcedit/issues/2
## Reasons to favor puppet syntax over pascal for .pp files:

I will not argue whether in 2014 pascal is still being used or not,
and to what extent. I would like to point out that whoever is using
any variation of pascal for any serious development, is probably not
using 'mcedit' as an IDE. Instead, this development activity is
probably situated in nice, fully-featured graphical IDE.

Platform wise, judging by the weekly download statistics of FreePascal
at SourceForge, at this moment more than 80% of all downloads are for
Windows environents. On the other hand Midnight Commander is Linux
based. Considering this in combination with speculation from the
previous paragraph, there seems to be only a small chance that 'mcedit'
is being used as an editor of choice for pascal development.

On the contrary, puppet currently certainly does have wider audience
amongst sysadmins, who regularly use Linux. Puppet master requires a
server (usually) and quite a few sysadmins edit files directly on
servers that act as puppet masters for puppet development
environments.

When accessing these servers remotely, mcedit comes very handy for
quick edits here and there. Also keep in mind that puppet files are
normally not very large. Therefore actual IDE would be an overkill and
'mcedit' does the job quite ok, but syntax highlighting would be
much appreciated.

To sum it up:
If we speculate about the _actual_ usage of 'mcedit', I do believe that
a wider audience would benefit by having .pp files highlighted by
default with puppet syntax definition instead of pascal syntax
definition.

### CODE REVIEW

**Comment by mrd000** [comment]
Just wanted to give my vote for this one.

**Comment by slavaz** [comment]
The pull request was merged into master.
https://www.midnight-commander.org/ticket/3266#no3

**Comment by ciampix** [comment]
Anyway we could leave support for Pascal files with the .pas extension, since this is a more common default for FreePascal and Lazarus Object Pascal IDE platform files.
